odl/space/fspace.py

```diff
@@ -154,199 +182,293 @@ class FunctionSetVector(Operator):
+
+        kwargs : {'vec_bounds_check'}
+            'bounds_check' : bool
+                Whether or not to check if all input points lie in
+                the function domain. For vectorized evaluation,
+                this requires the domain to implement
+                `contains_all`.
+
+                Default: `True`
 
         Returns
         -------
-        out : `FunctionSet.range` element or array of elements
-            Result of the function evaluation.
-        """
-        if x in self.domain:
-            # single value list: f(0, 1, 2)
-            pass
-        elif x[0] in self.domain:
-            # single array: f([0, 1, 2])
-            pass
-        else:  # Try vectorization
-            if not isinstance(self.domain, IntervalProd):
-                raise TypeError('vectorized evaluation only possible for '
-                                '`IntervalProd` domains.')
-            # Vectorization only allowed in this case
-
-            # First case: (N, d) array of points, where d = dimension
-            if (isinstance(x[0], np.ndarray) and
-                    x[0].ndim == 2 and
-                    x[0].shape[1] == self.domain.ndim):
-                min_coords = np.min(x[0], axis=0)
-                max_coords = np.max(x[0], axis=0)
-
-            # Second case: d meshgrid type arrays
-            elif (len(x) == self.domain.ndim and
-                    all(isinstance(vec, np.ndarray) for vec in x)):
-                min_coords = [np.min(vec) for vec in x]
-                max_coords = [np.max(vec) for vec in x]
+        out : range element or array of elements
+            Result of the function evaluation
 
-            else:
-                raise TypeError('input is neither an element of the '
-                                'function domain {!r} nor an array or '
-                                'meshgrid-type coordinate list.'
-                                ''.format(self.domain))
+        Raises
+        ------
+        TypeError
+            If `x` is not a valid vectorized evaluation argument
+
+            If `out` is not a range element or a `numpy.ndarray`
+            of range elements
 
-            if (min_coords not in self.domain or
-                    max_coords not in self.domain):
+        ValueError
+            If evaluation points fall outside the valid domain
+        """
+        vec_bounds_check = kwargs.pop('vec_bounds_check', True)
+        if vec_bounds_check and not hasattr(self.domain, 'contains_all'):
+            raise AttributeError('vectorized bounds check not possible for '
+                                 'domain {}, missing `contains_all()` '
+                                 'method.'.format(self.domain))
+
+        # Check for input type and determine output shape
+        if is_valid_input_array(x, self.domain.ndim):
+            out_shape = out_shape_from_array(x)
+            scalar_out = False
+            # For 1d, squeeze the array
```

Comment:
It checks if, in 1D, we have an array shape of `(N,)` or `(1, N)`, which are both valid but the latter needs to be squeezed.